table/server/master/src/main/java/alluxio/master/table/transform/TransformManager.java

```diff
@@ -231,14 +204,41 @@ public long execute(String dbName, String tableName, TransformDefinition definit
     return jobId;
   }
 
+  /**
+   * @param jobId the job ID
+   * @return the job information
+   */
+  public Optional<TransformJobInfo> getTransformJobInfo(long jobId) {
+    TransformJobInfo job = mState.getRunningJob(jobId);
+    if (job == null) {
+      job = mJobHistory.getIfPresent(jobId);
+    }
+    return job == null ? Optional.empty() : Optional.of(job);
+  }
+
+  /**
+   * @return all transformation jobs, including the running jobs and the kept finished jobs,
+   *    sorted by job ID in increasing order
+   */
+  public List<TransformJobInfo> getAllTransformJobInfo() {
+    List<TransformJobInfo> jobs = Lists.newArrayList(mJobHistory.asMap().values());
+    jobs.addAll(mState.getRunningJobs());
+    jobs.sort(Comparator.comparing(TransformJobInfo::getJobId));
+    return jobs;
+  }
+
+  @Override
+  public Journaled getDelegate() {
+    return mState;
+  }
+
   /**
    * Periodically polls the job service to monitor the status of the running transformation jobs,
    * if a transformation job succeeds, then update the Table partitions' layouts.
    */
   private final class JobMonitor implements HeartbeatExecutor {
-    private void completePartitionTransformation(String dbName, String tableName, TransformJob job)
-        throws IOException {
-      Table table = mCatalog.getTable(dbName, tableName);
+    private void updatePartition(TransformJobInfo job) throws IOException {
+      Table table = mCatalog.getTable(job.getDb(), job.getTable());
```

Comment:
could you move this update to inside `mCatalog`? locking is done in Catalog right now, and this is accessing unlocked Table object.  The future plan is to make mCatalog only return TableView object and DbView objects. All updates has to go through catalog.